### ISSUE TITLE
[Bug]: Terraform crashes with aws_wafv2_web_acl_logging_configuration updates

### ISSUE THREAD

**Issue by xuanit** [hashicorp/terraform-provider-aws#33926]
### Terraform Core Version

0.12.31

### AWS Provider Version

3.76.1

### Affected Resource(s)

aws_wafv2_web_acl_logging_configuration

### Expected Behavior

Terraform should be able to plan the change.

### Actual Behavior

Terraform crashes with the errors below when updating the resource.

### Relevant Error/Panic Output Snippet

```shell
panic: inconsistent set element types (cty.Object(map[string]cty.Type{"action_condition":cty.Tuple([]cty.Type{cty.Object(map[string]cty.Type{"action":cty.String})}), "label_name_condition":cty.EmptyTuple}) then cty.Object(map[string]cty.Type{"action_condition":cty.EmptyTuple, "label_name_condition":cty.Tuple([]cty.Type{cty.Object(map[string]cty.Type{"label_name":cty.String})})}))
goroutine 5095 [running]:
github.com/zclconf/go-cty/cty.SetVal(0xc002314800, 0x2, 0x2, 0x1bd1de0, 0xc0030a5770, 0x246ca00, 0x246caa0)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.2.1/cty/value_init.go:258 +0x76d
github.com/hashicorp/terraform/plans/objchange.setElementCompareValue(0xc001ef7b30, 0x246caa0, 0xc00347a7b0, 0x1bd1de0, 0xc0030a5e30, 0x246ca00, 0xc00347a7b0, 0x1bd1de0, 0xc0030a5e30, 0x6)
	/home/circleci/project/project/plans/objchange/objchange.go:358 +0x1149
github.com/hashicorp/terraform/plans/objchange.setElementCompareValues(0xc001ef7b30, 0x246cae0, 0xc001d00be0, 0x1e73ce0, 0xc0024c7140, 0x0, 0x246cae0, 0xc0011de600, 0x1e73ce0)
	/home/circleci/project/project/plans/objchange/objchange.go:300 +0x1b3
github.com/hashicorp/terraform/plans/objchange.proposedNewObject(0xc001ef7a40, 0x246caa0, 0xc00347a8f0, 0x1bd1de0, 0xc001eb6660, 0x246caa0, 0xc000e67e20, 0x1bd1de0, 0xc0045b4180, 0x246c9a0, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:237 +0x1892
github.com/hashicorp/terraform/plans/objchange.ProposedNewObject(0xc001ef7a40, 0x246caa0, 0xc00347a8f0, 0x1bd1de0, 0xc001eb6660, 0x246caa0, 0xc000e67e20, 0x1bd1de0, 0xc0045b4180, 0xc00347a8f0, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:42 +0x127
github.com/hashicorp/terraform/plans/objchange.proposedNewObject(0xc001e9dfb0, 0x246caa0, 0xc00347a928, 0x1bd1de0, 0xc001eb7590, 0x246caa0, 0xc000e67e90, 0x1bd1de0, 0xc0045b48a0, 0x20, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:145 +0xb4d
github.com/hashicorp/terraform/plans/objchange.ProposedNewObject(0xc001e9dfb0, 0x246caa0, 0xc00347a928, 0x1bd1de0, 0xc001eb7590, 0x246caa0, 0xc000e67e90, 0x1bd1de0, 0xc0045b48a0, 0xc0023ed9e0, ...)
	/home/circleci/project/project/plans/objchange/objchange.go:42 +0x127
github.com/hashicorp/terraform/terraform.(*EvalDiff).Eval(0xc0021c4000, 0x24a2960, 0xc00223dc70, 0x2, 0x2, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval_diff.go:164 +0x6d6
github.com/hashicorp/terraform/terraform.EvalRaw(0x2427320, 0xc0021c4000, 0x24a2960, 0xc00223dc70, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/eval.go:57 +0x131
github.com/hashicorp/terraform/terraform.(*EvalSequence).Eval(0xc00271af80, 0x24a2960, 0xc00223dc70, 0x2, 0x2, 0xf3e66d, 0x2427740)
	/home/circleci/project/project/terraform/eval_sequence.go:20 +0xfd
github.com/hashicorp/terraform/terraform.EvalRaw(0x2427660, 0xc00271af80, 0x24a2960, 0xc00223dc70, 0x1bc2480, 0x3535782, 0x1b25bc0, 0xc003ad2bb0)
	/home/circleci/project/project/terraform/eval.go:57 +0x131
github.com/hashicorp/terraform/terraform.Eval(0x2427660, 0xc00271af80, 0x24a2960, 0xc00223dc70, 0xc00271af80, 0x2427660, 0xc00271af80, 0x0)
	/home/circleci/project/project/terraform/eval.go:35 +0x4d
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x1e76ba0, 0xc002e8d9e0, 0x0, 0x0, 0x0)
	/home/circleci/project/project/terraform/graph.go:90 +0xf40
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc00238c900, 0x1e76ba0, 0xc002e8d9e0, 0xc001a52640)
	/home/circleci/project/project/dag/walk.go:392 +0x353
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/home/circleci/project/project/dag/walk.go:314 +0xa9b
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.
When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.
SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker
```

### Terraform Configuration Files

AWS provider version:  ~> 3.76
Terraform version: 0.12.31

```terraform
resource "aws_wafv2_web_acl_logging_configuration" "waf2" {
  count                   = var.logging_enabled ? 1 : 0
  log_destination_configs = var.logging_locations
  resource_arn            = join("", aws_wafv2_web_acl.main.*.arn)
  dynamic "redacted_fields" {
    for_each = var.logging_redacted_headers
    content {
      single_header {
        name = redacted_fields.value
      }
    }
  }

  logging_filter {
    default_behavior = "DROP"

    filter {
      behavior = "DROP"

      condition {
        label_name_condition {
          label_name = "awswaf:managed:aws:core-rule-set:SizeRestrictions_Body"
        }
      }
      condition {
        action_condition {
          action = "EXCLUDED_AS_COUNT"
        }
      }

      requirement = "MEETS_ALL"
    }

    filter {
      behavior = "KEEP"

      condition {
        action_condition {
          action = "EXCLUDED_AS_COUNT"
        }
      }

      requirement = "MEETS_ANY"
    }

    filter {
      behavior = "KEEP"

      condition {
        action_condition {
          action = "BLOCK"
        }
      }

      requirement = "MEETS_ANY"
    }

    filter {
      behavior = "KEEP"

      condition {
        action_condition {
          action = "COUNT"
        }
      }

      requirement = "MEETS_ANY"
    }
  }

  lifecycle {
    ignore_changes = ["logging_filter"]
  }
}
```

### Steps to Reproduce

1) Config a WAF log resource like the above
2) Apply to create the resource
3) Plan again and terraform will crash

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Hey @xuanit 👋 Thank you for taking the time to raise this! I notice that you're using quite outdated versions of Terraform and the AWS Provider. If possible, can you test with more recent versions to verify whether the issue persists?

**Comment by xuanit** [comment]
Thanks! It was fixed in terraform version 1.6.3.
